marimbabot_speech/src/speech2text.py

```diff
@@ -56,7 +52,9 @@ def handle_audio_buffer(self, req):
 
 	# To address the way how to unpack the data for whisper
 	def unpack_stream(self, data):
-		return np.array(struct.unpack(f"{int(len(data) / 2)}h", bytes(data)), dtype=float) / 526
+		# return np.array(struct.unpack(f"{int(len(data) / 2)}h", bytes(data)), dtype=float) / 526
+		return np.frombuffer(data, dtype=np.float64)/526
```

Comment:
I think this doesn't work. I casted the bytes to ints and then converted the ints to floats. This changed the binary representation. Loading them directly as floats will result in wrong values.